tests/test_sysadmin.py

```diff
@@ -0,0 +1,387 @@
+        return self.client.post(reverse("sysadmin_courses"), post_dict)
+
+    def _rm_edx4edx(self):
+        """Deletes the sample course from the XML store"""
+        def_ms = modulestore()
+        course_path = "{}/edx4edx_lite".format(os.path.abspath(settings.DATA_DIR))
+        try:
+            # using XML store
+            course = def_ms.courses.get(course_path, None)
+        except AttributeError:
+            # Using mongo store
+            course = def_ms.get_course(CourseLocator("MITx", "edx4edx", "edx4edx"))
+
+        # Delete git loaded course
+        response = self.client.post(
+            reverse("sysadmin_courses"),
+            {
+                "course_id": str(course.id),
+                "action": "del_course",
+            },
+        )
+        self.addCleanup(self._rm_glob, f"{course_path}_deleted_*")
+
+        return response
+
+    def _rm_glob(self, path):
+        """
+        Create a shell expansion of passed in parameter and iteratively
+        remove them.  Must only expand to directories.
+        """
+        for path in glob.glob(
+            path
+        ):  # lint-amnesty, pylint: disable=redefined-argument-from-local
+            shutil.rmtree(path)
+
+    def _mkdir(self, path):
+        """
+        Create directory and add the cleanup for it.
+        """
+        os.mkdir(path)
+        self.addCleanup(shutil.rmtree, path)
+
+
+@override_settings(
+    MONGODB_LOG=TEST_MONGODB_LOG,
+    GIT_REPO_DIR=settings.TEST_ROOT / f"course_repos_{uuid4().hex}",
+)
+@unittest.skipUnless(
+    settings.FEATURES.get("ENABLE_SYSADMIN_DASHBOARD"),
+    "ENABLE_SYSADMIN_DASHBOARD not set",
+)
+class TestSysAdminMongoCourseImport(SysadminBaseTestCase):
+    """
+    Check that importing into the mongo module store works
+    """
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete mongo log entries after test."""
+        super().tearDownClass()
+        try:
+            mongoengine.connect(TEST_MONGODB_LOG["db"])
+            CourseImportLog.objects.all().delete()
+        except mongoengine.connection.ConnectionFailure:
+            pass
+
+    def _setstaff_login(self):
+        """
+        Makes the test user staff and logs them in
+        """
+
+        self.user.is_staff = True
+        self.user.save()
+
+        self.client.login(username=self.user.username, password="foo")
+
+    @pytest.mark.skip(reason="FIXME: Refactor needed for the test")
+    def test_missing_repo_dir(self):
+        """
+        Ensure that we handle a missing repo dir
+        """
+
+        self._setstaff_login()
+
+        if os.path.isdir(settings.GIT_REPO_DIR):
+            shutil.rmtree(settings.GIT_REPO_DIR)
+
+        # Create git loaded course
+        response = self._add_edx4edx()
+        self.assertContains(
+            response, Text(str(GitImportErrorNoDir(settings.GIT_REPO_DIR)))
+        )
+
+    @pytest.mark.skip(reason="FIXME: Refactor needed for the test")
```

Comment:
Do we have a ticket on the backlog for these FIXMEs?

Reply:
Yes, I've created a ticket for these https://github.com/mitodl/edx-sysadmin/issues/20.